### PR TITLE
feat(core): add `gr_cost` syscall

### LINKED ISSUE #2637: Ability to get various costs in user space (syscall)
### Problem to Solve

get waitlist cost, mailbox threshold, etc

### Possible Solution

`gr_cost(Enum::{Waitlist, MailboxThreshold, ...})`

### Notes

_No response_

### PR DESCRIPTION
**Pull request by tltsutltsu**
Resolves #2637.

- [x] add syscall
- [x] add example to test that syscall works
- [x] switch to enum input for syscall
- [ ] update weights

### CODE REVIEW

**Review by breathx — CHANGES_REQUESTED** [review]
syscall(i32) -> ptr err_len (u32 bytes) + cost (u128 bytes)

**Review by NikVolf — CHANGES_REQUESTED** [review]
please use ids instead of strings in ffi

**Comment by StackOverflowExcept1on** [comment]
Did you forget about `core-backend/wasmi/src/funcs_tree.rs`?

File: core-backend/common/src/funcs.rs [hunk 1]
```diff
@@ -429,6 +429,25 @@ where
             .map_err(Into::into)
     }
 
+    #[host(fallible, cost = RuntimeCosts::Cost, err = ErrorBytes)]
+    pub fn cost(
+        ctx: &mut R,
+        cost_name_ptr: u32,
```
Comment (breathx) [inline core-backend/common/src/funcs.rs:435]: what is these names etc?
just match num -> cost type

**Review by breathx — CHANGES_REQUESTED** [review]

**Comment by breathx** [comment]
To be reopened once work started again

**Pull request closed without merging**